api/types/server.go

```diff
@@ -83,6 +83,8 @@ type Server interface {
 
 	// GetCloudMetadata gets the cloud metadata for the server.
 	GetCloudMetadata() *CloudMetadata
+	// GetAWSInfo returns the AWSInfo for the server.
+	GetAWSInfo() *AWSInfo
```

Comment:
Why the dedicated method when we have GetCloudMetadata already?

Reply:
If I had to guess just a convenience method to avoid checking for "CloudMetadata == nil" on the caller side?

Reply:
Exactly
I had to do a nil check in two places, so it just seemed easier to create a function the Server interface and re-use it in those places

Reply:
Wouldn't we get the exact same behavior with `.GetCloudMetadata().GetAWS()`?

Reply:
`.GetCloudMetadata()` returns nil for some nodes, so that would panic

Reply:
TIL :pray: